aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -762,6 +810,31 @@ private int createMultipartUploadRecords(String bucket, String key, File file, O
         return dbUtil.bulkInsertTransferRecords(valuesArray);
     }
 
+    private File writeInputStreamToFile(InputStream inputStream) throws IOException {
+        if (inputStream == null) {
+            throw new IllegalArgumentException("Invalid inputStream: " + inputStream);
+        }
+
+        // Saves the data as a file in the temporary directory
+        File file = File.createTempFile("aws-s3", ".tmp");
+        if (!file.isFile()) {
```

Comment:
Hmm, why would the file not be a file?  Is this check necessary?

Reply:
Sorry, it should be `if (!file.isDirectory())`

Reply:
Hmm, in what scenario would it be a directory?  I'm not sure we need to check for that, since we just created it, as a File, not a directory.

Reply:
I just want to make sure that the file is created successfully. If not, throw an exception. Yeah we shouldn't use isDirectory(). I'm wrong😭

Reply:
Got it.  `File.createTempFile` will throw an `IOException` if there's any issue creating it, so that's how you can tell if it was successful.  You aren't catching that exception so it will just be thrown back to the customer, which I think is a reasonable approach.

Reply:
Make sense. I'll remove this check.